### PR TITLE
Mayor Webpack rework

### PR DESCRIPTION
**Pull request by juspky**
**Switched from ts-loader to awesome-typescript-loader**
Webpack notifier is fixed now
Performance is the same

**Removed vendor.js**
vendor.js is discouraged since webpack 4.
https://webpack.js.org/concepts/entry-points/#separate-app-and-vendor-entries
Webpack does chunk splitting internally now
https://webpack.js.org/plugins/split-chunks-plugin/#optimizationsplitchunks
Plus this removes the complexity of that build webpack build dll thing

**yarn watch is now called yarn start**
In the process of cleaning up package.json from vendor.js I renamed watch to start to match the usual react app naming of those commands.

**Polyfills are now loaded in the webpack entry from polyfills.js**
This gets rid of one dependency and provides a simple point to add new polyfills

**Updated Framework & Extensions**

**Update to typescript 3.3**

**Updated serveral other dependencies in package.json**

**Updated vscode launch configs**

### CODE REVIEW

**Comment by olmobrutall** [comment]
Wow, big changes here. Scary but exciting. 

We have to wait for this to get published https://github.com/jquense/react-widgets/issues/915 

In the meantime I'll check this changes. 

Thanks 😄

**Comment by olmobrutall** [comment]
I've done some benchmarking, it's much slower now 😢 

|   	|   Fresh Start |  change in Home.tsx  |  
|---	|---	|---	|
|  ts-loader 	|   29s	|   2s	|   
|   awesome-typescript-loader	|   1m 5s	|   12s	|

**Comment by juspky** [comment]
For me a change in home.tsx took way less

I'll see if I can get it to run faster

**Comment by olmobrutall** [comment]
Maybe split the webpack 4 change from the awesome typescript loader? 

Also, what’s the point for calling it start instead of watch and the changes in tsconfig? They look like the scars a of the fight  with babel

**Comment by juspky** [comment]
So I figured out is the cause of the _transpileOnly: false_ is the cause of being slow.
If I set it to _true_ (which it was with ts-loader too) its twice as fast but the harmony linking errors are back since this messes with module exports somehow.

**useTranspileModule: true**
Build: 52s

**useTranspileModule: false**
Build: 120s

**HardSourceWebpackPlugin**
First build: 122s
Second build (after rm -rf dist): 3.5s
Change In Home.tsx:
3.7s
2.8s
2.9s
1.6s
2.3s

So if you can bear with the first build being 60sec slower you'll get rid of the errors.

The TSConfig changes seemed reasonable you can see what they to here https://www.typescriptlang.org/docs/handbook/compiler-options.html

I decided to rename the script to start because this is what its usually called in other projects, we can change it back to watch if you want.

File: Southwind.React/webpack.config.js [hunk 1]
```diff
@@ -1,91 +1,103 @@
...
+        watch: "App",
+        silent: true,
+      }),
+      new HardSourceWebpackPlugin(),
+      // Ignore all locale files of moment.js
+      new webpack.IgnorePlugin(/^\.\/locale$/, /moment$/),
+      new AssetsPlugin({
+        path: path.join(__dirname, "wwwroot", "dist")
+      }),
+      new WebpackNotifierPlugin({ title: 'Signum', excludeWarnings: true, alwaysNotify: true }),
```
Comment (olmobrutall) [inline Southwind.React/webpack.config.js]: Is this line not enough?

**Comment by olmobrutall** [comment]
I don’t really know what to do with this PR, there are so many changes and the benefits are not clear to me. Is still slower ?

**Comment by juspky** [comment]
split chunks now bundles the same vendors as before

see
https://webpack.js.org/plugins/split-chunks-plugin/#split-chunks-example-3

**Comment by juspky** [comment]
Any doubts about the polyfills?

If no I guess the only thing left would be the ts-loader.

**Pull request closed without merging**